### PR TITLE
WIP: Add hdf4

### PR DESCRIPTION
**Pull request by groutr**
Related to #108.

hdf4 depends on #119

### CODE REVIEW

**Comment by groutr** [comment]
Removed bison because of #119

File: recipes/hdf4/build.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    make || return 1;
+    make install || return 1;
+
+    rm -rf ${PREFIX}/share/hdf4_examples;
+
+    return 0;
+}
+
+case ${ARCH} in
+    'Linux')
```
Comment (pelson) [inline recipes/hdf4/build.sh]: Just wondering if this is overkill? The two functions look very similar except for the CC and CXX flags...

Comment (groutr) [inline recipes/hdf4/build.sh]: Agreed.  Even the CC and CXX flags I don't think are entirely necessary.

**Comment by groutr** [comment]
With bison merged, these CI builds need to be re-run.

**Comment by pelson** [comment]
Can't remember if a power cycle does that or not. No harm in trying :smile:

**Comment by pelson** [comment]
Linux failure: 

```
Package: hdf4-4.2.11-0
source tree in: /opt/conda/conda-bld/work/hdf-4.2.11
+ export 'CFLAGS= -pipe -O2 -fPIC -I/opt/conda/envs/_build/include'
+ CFLAGS=' -pipe -O2 -fPIC -I/opt/conda/envs/_build/include'
+ export 'CXXFLAGS= -pipe -O2 -fPIC -I/opt/conda/envs/_build/include'
+ CXXFLAGS=' -pipe -O2 -fPIC -I/opt/conda/envs/_build/include'
+ export 'LDFLAGS=-L/opt/conda/envs/_build/lib '
+ LDFLAGS='-L/opt/conda/envs/_build/lib '
+ chmod +x configure
+ ./configure --disable-static --enable-linux-lfs --with-ssl --with-zlib --with-jpeg --disable-netcdf --disable-fortran --prefix=/opt/conda/envs/_build
configure: WARNING: unrecognized options: --enable-linux-lfs, --with-ssl
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
/opt/conda/conda-bld/work/hdf-4.2.11/bin/missing: Unknown `--is-lightweight' option
Try `/opt/conda/conda-bld/work/hdf-4.2.11/bin/missing --help' for more information
configure: WARNING: 'missing' script is too old or missing
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking shell variables initial values... done
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking if basename works... yes
checking if xargs works... yes
checking for config x86_64-unknown-linux-gnu... no
checking for config x86_64-unknown-linux-gnu... no
checking for config unknown-linux-gnu... no
checking for config unknown-linux-gnu... no
checking for config x86_64-linux-gnu... no
checking for config x86_64-linux-gnu... no
checking for config x86_64-unknown... no
checking for config linux-gnu... found
checking whether make sets $(MAKE)... (cached) yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking whether ln -s works... yes
checking for ar... ar
checking for bison... bison -y
checking for flex... no
checking for lex... no
configure: error: cannot find lex utility
```

File: recipes/hdf4/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  fn: hdf-4.2.11.tar.bz2
+  url: http://www.hdfgroup.org/ftp/HDF/HDF_Current/src/hdf-4.2.11.tar.bz2
+
+
+requirements:
+  build:
+    - jpeg   8d
+    - zlib   1.2.*
+    - cmake  # [win]
+    - bison  # [unix]
```
Comment (jakirkham) [inline recipes/hdf4/meta.yaml]: Add `flex` for UNIX only, as well.

**Comment by jakirkham** [comment]
> Can't remember if a power cycle does that or not. No harm in trying :smile:

It hurts AppVeyor. The rest normally behave ok.

**Comment by groutr** [comment]
I'm not sure how to reduce output from a build, but Travis is failing because the log output is too long.

**Comment by jakirkham** [comment]
Yeah, there are some fun constraints on Travis. Unfortunately, there is no way to override these. Maybe you could try to find some options to quiet various steps.

**Comment by jakirkham** [comment]
Looks like the Python 3.5 tests are hanging on AppVeyor.

**Comment by ocefpaf** [comment]
The Python 3.5 build on Windows just hangs, so I canceled them to let AppVeyor free :wink: 

@gillins can you take a look at this? Maybe it needs the same magic you did to `gdal` and `geos`.

**Comment by gillins** [comment]
Yep it does indeed hang with Python 3.5 on Windows. Attached is patch to fix the bit of cmake that hangs (see https://github.com/conda/conda-recipes/issues/568)
[HDFTests.c.patch.txt](https://github.com/conda-forge/staged-recipes/files/207488/HDFTests.c.patch.txt)

Also a problem is there isn't support for VC features, so needs this in the 'build' section:
    features:
        - vc9  # [win and py27]
        - vc10  # [win and py34]
        - vc14  # [win and py35]

and python as a build requirement. 
I'm not sure if I can make these changes directly to the PR? Or it is easiest for @groutr to do it?

**Comment by groutr** [comment]
That's my bad.  I thought I had already added the vc feature selectors.

I'll look at the patch today.

**Comment by gillins** [comment]
Do you still get an build error with this?

**Comment by ocefpaf** [comment]
> Do you still get an build error with this?

The build was "contaminated" by some `jupyter` packages that lingered around due to a failure in the feedstock creation. But the patches seems to work! I re-started `AppVeyor` so we can take a closer look.

@groutr should we get a `zlib` recipe going before we merge this one?

**Comment by ocefpaf** [comment]
Windows is green!

**Comment by groutr** [comment]
Sure, I can get a zlib package going.

**Pull request closed without merging**